agent/engine/docker_task_engine_test.go

```diff
@@ -3281,18 +3291,27 @@ func TestMonitorExecAgentRunning(t *testing.T) {
 		if tc.simulateBadContainerId {
 			testTask.Containers[0].RuntimeID = ""
 		}
-		if tc.containerStatus == apicontainerstatus.ContainerRunning && !tc.simulateBadContainerId {
-			execCmdMgr.EXPECT().RestartAgentIfStopped(dockerTaskEngine.ctx, dockerTaskEngine.client, testTask, testTask.Containers[0], testContainerId).
+		if tc.containerStatus == apicontainerstatus.ContainerRunning && tc.willRestart {
```

Comment:
I was having issues reading the logic here because I wasn't seeing that this line is the only lines that change are # 3297 and # 3303

Would it be better to express this whole block as:

```golang
restartStatus := execcmd.Restarted
if !tc.willRestart {
    restartStatus := execcmd.NotRestarted
}

if tc.containerStatus == apicontainerstatus.ContainerRunning && !tc.simulateBadContainerId {
    execCmdMgr.EXPECT().RestartAgentIfStopped(dockerTaskEngine.ctx, dockerTaskEngine.client, testTask,
        testTask.Containers[0], testContainerId).
        Return(restartStatus, nil).
        Times.(1)
}
```

Reply:
or even better, we could put the `restartStatus` in the test table, instead of `willRestart`